modules/apigee_edge_actions/src/EventSubscriber/AppCredentialEventSubscriber.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+/**
+ * Copyright 2020 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge_actions\EventSubscriber;
+
+use Drupal\apigee_edge_actions\Event\EdgeEntityEventEdge;
+use Drupal\apigee_edge\Entity\AppInterface;
+use Drupal\apigee_edge\Event\AppCredentialAddApiProductEvent;
+use Drupal\apigee_edge\Event\AppCredentialDeleteApiProductEvent;
+use Drupal\Component\Plugin\Exception\PluginException;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Logger\LoggerChannelInterface;
+use Drupal\Core\Session\AccountInterface;
+use Symfony\Component\EventDispatcher\Event;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Proxy for Apigee Edge app credential events.
```

Comment:
A little more detail like this could help explain class:
```suggestion
 * Events for an API Product being added to an app already exist. This event subscriber is a proxy for Apigee Edge app credential events. 
```